Update documentation

ala-mode opened the pull request:
Updating documentation.

An invitation to discussion or expansion.

ala-mode committed:
update docs

ala-mode committed:
begin cargo make documentation

ala-mode committed:
remove get_subtree_roots testing documentation

ala-mode committed:
Merge branch 'dev' into update_documentation

ala-mode committed:
Merge branch 'dev' into update_documentation

nachog00 committed:
Merge branch 'dev' into update_documentation

idky137 committed:
Merge branch 'dev' into update_documentation

AloeareV reviewed (changes requested):
Sorry to get to this so late! This gets rid of a good deal of out-of-date documentation, which is very nice. Currently `cargo nextest run` isn't sufficient for the majority of the tests for me, there's the step of getting the tests access to the binaries (which I have done by symlinking zainod, zebrad, and zcashd into a directory which I point the environment variable `TEST_BINARIES_DIR`). I think cargo make is supposed to solve this problem in some manner, but I don't understand the details. Want to take a swing at documenting this element of running the tests? 

I'd also be okay merging this as-is on the grounds that it's less incorrect

ala-mode committed:
Merge branch 'dev' into update_documentation

ala-mode committed:
add env var instuctions, elaborate cargo make

ala-mode commented:
@AloeareV - I made some adjustments to the doc, thank you for your review.

I added a step about adding the the binaries directory path to `$PATH` (or to `$TEST_BINARIES_DIR` which you said worked for you).

I also added a little context about `cargo make`, but I don't really understand that workflow, so it's a pointer, and a bit of placeholder.

idky137 committed:
Merge branch 'dev' into update_documentation

idky137 commented on docs/testing.md:
```diff
@@ -6,34 +6,20 @@
-To run client rpc test `get_subtree_roots_orchard`:
-1) sync Zebrad mainnet to at least 2 orchard shards
-2) copy the Zebrad mainnet `state` cache to `zaino/integration-tests/chain_cache/get_subtree_roots_orchard` directory.
+3) Build release binary `cargo build --release`
```
Thanks for updating this! (Way overdue!) Looks good!

One more line I'm fairly certain we can remove (maybe someone else should double check this other than me) is `cargo build --release`.

This used to be a requirement for the `Infrastructure::client_rpc_tests` but as these have been removed (they are now only held in Infrastructure) we should no longer have to run this.

ala-mode replied:
Hello @idky137 !

Sorry it took me so long to circle back to this, thank you for your review. I think you were right, and I made some more changes to the docs.

I've updated the branch, and am running some tests locally to make sure everything works (without `zainod` in `/target/...` or in `test_binaries/bin` as expected.

ala-mode committed:
Merge branch 'dev' into update_documentation

ala-mode committed:
remove instruction to build zainod

ala-mode commented:
Hey @AloeareV I think you could make quick work of reviewing this too!

zancas committed:
Merge branch 'dev' into update_documentation

ala-mode committed:
add .env.testing-artifacts info

zancas committed:
Merge branch 'dev' into update_documentation

AloeareV reviewed (approved):
Big improvement to docs! I wish I'd merged this weeks ago, but the second best time is now I suppose :)
